fdbserver/workloads/WatchesSameKeyCorrectness.actor.cpp

```diff
@@ -57,13 +61,18 @@ struct WatchesSameKeyWorkload : TestWorkload {
 		return ok;
 	}
 
+	static int getId() {
+		return uid++;
+	}
+
 	ACTOR static Future<Void> setKeyRandomValue(Database cx, Key key, Optional<Value> val) {
 		state ReadYourWritesTransaction tr(cx);
 		loop {
 			try {
-				if (!val.present())
-					val = Value(deterministicRandom()->randomUniqueID().toString());
-				tr.set(key, val.get());
+				Value valS;
```

Comment:
Main issue was that i was re-assigning to the parameter `val` (in the case of an error) instead of a copy